tests/components/netatmo/test_init.py

```diff
@@ -0,0 +1,199 @@
+    ) as mock_webhook:
+        mock_auth.return_value.post_request.side_effect = fake_post_request
+        assert await async_setup_component(hass, "netatmo", {})
+
+    await hass.async_block_till_done()
+
+    assert mock_auth.call_count == 1
+    assert mock_impl.call_count == 1
+    assert mock_webhook.call_count == 1
+
+    assert config_entry.state == config_entries.ENTRY_STATE_LOADED
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+
+    assert hass.states.get("climate.netatmo_livingroom").state == "auto"
+
+    for config_entry in hass.config_entries.async_entries("netatmo"):
+        await hass.config_entries.async_remove(config_entry.entry_id)
+
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) == 0
+
+    assert hass.states.get("climate.netatmo_livingroom") is None
+
+
+async def test_setup_component_with_config(hass, config_entry):
+    """Test setup of the netatmo component with dev account."""
+    with patch(
+        "homeassistant.components.netatmo.api.ConfigEntryNetatmoAuth"
+    ) as mock_auth, patch(
+        "homeassistant.helpers.config_entry_oauth2_flow.async_get_config_entry_implementation",
+    ) as mock_impl, patch(
+        "homeassistant.components.webhook.async_generate_url"
+    ) as mock_webhook:
+        mock_auth.return_value.post_request.side_effect = fake_post_request
+        assert await async_setup_component(
+            hass, "netatmo", {"netatmo": {"client_id": "123", "client_secret": "abc"}}
+        )
+
+    await hass.async_block_till_done()
+
+    assert mock_auth.call_count == 1
+    assert mock_impl.call_count == 1
+    assert mock_webhook.call_count == 1
+
+    assert config_entry.state == config_entries.ENTRY_STATE_LOADED
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+
+    assert "camera" in hass.data
+    assert "climate" in hass.data
+    assert "light" in hass.data
+    assert "sensor" in hass.data
+
+
+async def test_setup_component_with_webhook(hass, entry):
+    """Test setup and teardown of the netatmo component with webhook registration."""
+    webhook_data = {
+        "user_id": "123",
+        "user": {"id": "123", "email": "foo@bar.com"},
+        "push_type": "webhook_activation",
+    }
+    async_dispatcher_send(
+        hass,
+        f"signal-{DOMAIN}-webhook-None",
+        {"type": None, "data": webhook_data},
+    )
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) > 0
+
+    for config_entry in hass.config_entries.async_entries("netatmo"):
+        await hass.config_entries.async_remove(config_entry.entry_id)
+
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) == 0
+
+    assert hass.states.get("climate.netatmo_livingroom") is None
+
+
+async def test_setup_without_https(hass, config_entry):
+    """Test if set up with cloud link and without https."""
+    hass.data["cloud"] = Mock()
```

Comment:
Don't modify hass.data directly. Patch something instead to have the desired outcome.